derive/Cargo.toml

```diff
@@ -19,3 +19,4 @@ quote = "1.0"
 syn = { version = "1.0", features = ["derive", "visit", "visit-mut", "extra-traits"] }
 proc-macro2 = "1.0"
 proc-macro-crate = "0.1.5"
+hashbrown = "0.9.1"
```

Comment:
It was actually a question I had: why do we make the `derive` crate no-std ([here](https://github.com/paritytech/scale-info/blob/master/derive/src/lib.rs#L15))? If it's fine to remove that constraint then `std::collection::HashSet` is obviously better than a new dependency.

The reason for wanting `HashSet` is to avoid adding duplicate bounds.

Reply:
Proc. macros are `std` by default. Dependencies like `serde` do not even support `no_std`. Only the code proc. macros generate might need to be `no_std` (as in our case).

Reply:
Ok, I removed https://github.com/paritytech/scale-info/blob/master/derive/src/lib.rs#L15 and switched to std's HashSet.